Mathlib/CategoryTheory/EpiMono.lean

```diff
@@ -242,4 +242,68 @@ instance {X Y : C} (f : X ⟶ Y) [hf : IsSplitEpi f] (F : C ⥤ D) : IsSplitEpi
 
 end
 
+section
+
+/-- When `f` is an isomorphism, `f ≫ g` is epic iff `g` is. -/
+@[simp]
+lemma epi_isIso_comp_iff {X Y Z : C} (f : X ⟶ Y) [IsIso f] (g : Y ⟶ Z) :
+    Epi (f ≫ g) ↔ Epi g := by
+  refine ⟨fun h ↦ ?_, fun h ↦ inferInstance⟩
+  simpa using (inferInstance : Epi (inv f ≫ f ≫ g))
+
+/-- When `g` is an isomorphism, `f ≫ g` is epic iff `f` is. -/
+@[simp]
+lemma epi_comp_isIso_iff {X Y Z : C} (f : X ⟶ Y) (g : Y ⟶ Z) [IsIso g] :
+    Epi (f ≫ g) ↔ Epi f := by
+  refine ⟨fun h ↦ ?_, fun h ↦ inferInstance⟩
+  simpa using (inferInstance : Epi ((f ≫ g) ≫ inv g ))
+
+/-- When `f` is an isomorphism, `f ≫ g` is monic iff `g` is. -/
+@[simp]
+lemma mono_isIso_comp_iff {X Y Z : C} (f : X ⟶ Y) [IsIso f] (g : Y ⟶ Z) :
+    Mono (f ≫ g) ↔ Mono g := by
+  refine ⟨fun h ↦ ?_, fun h ↦ inferInstance⟩
+  simpa using (inferInstance : Mono (inv f ≫ f ≫ g))
+
+/-- When `g` is an isomorphism, `f ≫ g` is monic iff `f` is. -/
+@[simp]
+lemma mono_comp_isIso_iff {X Y Z : C} (f : X ⟶ Y) (g : Y ⟶ Z) [IsIso g] :
+    Mono (f ≫ g) ↔ Mono f := by
+  refine ⟨fun h ↦ ?_, fun h ↦ inferInstance⟩
+  simpa using (inferInstance : Mono ((f ≫ g) ≫ inv g ))
+
+/-- For any natural transformation `f : F ⟶ G`, if `X` and `Y` are isomorphic, the component of
+`f` at `X` is epic iff the component of `f` at `Y` is. -/
+lemma NatTrans.epi_app_congr_iso {C D : Type*} [Category C] [Category D]
```

Comment:
As I am not completely use we absolutely need these lemmas (in my code above about `HToF`, I did not need it), I would suggest keeping these lemmas out of this PR.